homeassistant/components/asus_router/config_flow.py

```diff
@@ -0,0 +1,429 @@
+
+        errors = {}
+
+        if user_input:
+            # Check credentials and connection
+            result = await _async_check_connection(self.hass, self._configs, user_input)
+            # Show errors if any
+            if ERRORS in result:
+                errors[BASE] = result[ERRORS]
+            else:
+                # Saved the checked settings to the options
+                self._options.update(result[CONFIGS])
+                # Set unique ID obtained from the device during the check
+                await self.async_set_unique_id(result[UNIQUE_ID])
+                # Proceed to the next step
+                return await _async_process_step(self._steps, step_id)
+
+        user_input = self._options.copy()
+
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=_create_form_credentials(user_input),
+            errors=errors,
+        )
+
+    # Step Finish
+    async def async_step_finish(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Finish setup."""
+
+        return self.async_create_entry(
+            title=self._configs[CONF_HOST],
+            data=self._configs,
+            options=self._options,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Get the options flow."""
+
+        return AROptionsFlowHandler(config_entry)
+
+
+class AROptionsFlowHandler(OptionsFlow):
+    """Options flow for AsusRouter."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize options flow."""
+
+        self.config_entry = config_entry
+
+        self._configs: dict[str, Any] = self.config_entry.data.copy()
+        self._host: str = self._configs[CONF_HOST]
+        self._options: dict[str, Any] = self.config_entry.options.copy()
+
+        # Dictionary last_step: next_step
+        self._steps: dict[str, dict[str, Any]] = {
+            STEP_CREDENTIALS: {
+                METHOD: self.async_step_credentials,
+                NEXT: STEP_FINISH,
+            },
+            STEP_FINISH: {METHOD: self.async_step_finish},
+        }
+
+    async def async_step_init(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Options flow."""
+
+        return await self.async_step_credentials(user_input)
+
+    # Step #1 - credentials and SSL
+    async def async_step_credentials(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Credentials step."""
+
+        step_id = STEP_CREDENTIALS
+
+        errors = {}
+
+        if user_input:
+            self._options.update(user_input)
+            result = await _async_check_connection(
+                self.hass, self._configs, self._options
+            )
+            if ERRORS in result:
+                errors[BASE] = result[ERRORS]
```

Comment:
Hello and thanks for your feedback!

Yes, your suggestion is good. But this would require splitting the configuration for login/password (which will be handled by the re-auth flow and for which ConfigEntryAuthFailed can be raised properly) from the SLL / port settings as they are now.

Also, the credentials on the router do not expire as happens with some connected services and so on. The user needs to do this manually from the same menu as changing port/SSL - this was a part of the motivation to put them all together.

This can be implemented at the next step as a new feature and will help make the integration better. For now, I would prefer to keep the proposed code as small as possible, so the Core maintainers will be able to review it and approve it (if they find it correct and suitable).

Of course, if this plan is fine with you